p2panda-sync-next/src/test_utils.rs

```diff
@@ -0,0 +1,72 @@
+// SPDX-License-Identifier: MIT OR Apache-2.0
+
+use std::{collections::HashMap, convert::Infallible};
+
+use p2panda_core::{Body, Hash, Header, PrivateKey};
+use serde::{Deserialize, Serialize};
+
+use crate::log_sync::Logs;
+use crate::topic_log_sync::TopicLogMap;
+use crate::traits::TopicQuery;
+
+pub fn create_operation(
+    private_key: &PrivateKey,
+    body: &Body,
+    seq_num: u64,
+    timestamp: u64,
+    backlink: Option<Hash>,
+) -> (Header, Vec<u8>) {
+    let mut header = Header {
+        version: 1,
+        public_key: private_key.public_key(),
+        signature: None,
+        payload_size: body.size(),
+        payload_hash: Some(body.hash()),
+        timestamp,
+        seq_num,
+        backlink,
+        previous: vec![],
+        extensions: Some(()),
+    };
+    header.sign(private_key);
+    let header_bytes = header.to_bytes();
+    (header, header_bytes)
+}
```

Comment:
Soon we can use this: https://github.com/p2panda/p2panda/blob/stream-next/p2panda-core/src/test_utils.rs :-)